src/helper/service.cpp

```diff
@@ -54,8 +57,17 @@ service::ptr service::from_settings(const service_identifier &id,
     }
 
     std::chrono::milliseconds poll_interval{rc_settings.poll_interval};
-    auto rc_client = remote_config::client::from_settings(id, rc_settings);
+
+    // Create remote configs stuff
+    auto rc_service = std::make_shared<service_config>();
+    auto listener =
+        std::make_shared<remote_config::asm_features_listener>(rc_service);
+    std::vector<remote_config::product> products = {{"ASM_FEATURES", listener}};
+    auto rc_client = remote_config::client::from_settings(
+        id, rc_settings, std::move(products));
+
     return std::make_shared<service>(id, engine_ptr, std::move(rc_client),
+        std::move(rc_service),
```

Comment:
This can be instantiated inside the service on the constructor, there is no value to passing this in the constructor.

Reply:
I did it this way so I can assign the same service_config to the asm listener. Can you confirm you are ok with how it is? (names are changing)

Reply:
Ah right chicken-egg problem, sure leave it as is then.